Indicators/ImpliedVolatility.cs

```diff
@@ -175,32 +195,205 @@ public class ImpliedVolatility : OptionIndicatorBase
-            : this(name, option, new ConstantRiskFreeRateInterestRateModel(riskFreeRate), new ConstantDividendYieldModel(dividendYield), period, optionModel)
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, int period = 252)
+            : this(name, option, new ConstantRiskFreeRateInterestRateModel(riskFreeRate), new ConstantDividendYieldModel(dividendYield), optionModel, period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the ImpliedVolatility class
+        /// </summary>
+        /// <param name="option">The option to be tracked</param>
+        /// <param name="riskFreeRate">Risk-free rate, as a constant</param>
+        /// <param name="dividendYield">Dividend yield, as a constant</param>
+        /// <param name="optionModel">The option pricing model used to estimate IV</param>
+        /// <param name="period">The lookback period of historical volatility</param>
+        public ImpliedVolatility(Symbol option, decimal riskFreeRate = 0.05m, decimal dividendYield = 0.0m,
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, int period = 252)
+            : this($"IV({option.Value},{riskFreeRate},{dividendYield},{optionModel},{period})", option, riskFreeRate, dividendYield, optionModel, period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the ImpliedVolatility class
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="option">The option to be tracked</param>
+        /// <param name="mirrorOption">The mirror option for parity calculation</param>
+        /// <param name="riskFreeRateModel">Risk-free rate model</param>
+        /// <param name="dividendYieldModel">Dividend yield model</param>
+        /// <param name="optionModel">The option pricing model used to estimate IV</param>
+        /// <param name="period">The lookback period of historical volatility</param>
+        public ImpliedVolatility(string name, Symbol option, Symbol mirrorOption, IRiskFreeInterestRateModel riskFreeRateModel, IDividendYieldModel dividendYieldModel,
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, int period = 252)
+            : this(name, option, riskFreeRateModel, dividendYieldModel, optionModel, period)
+        {
+            SetMirrorOptionContract(mirrorOption);
+
+            SmoothingFunction = (impliedVol, mirrorImpliedVol) =>
+            {
+                if (Strike > UnderlyingPrice && Right == OptionRight.Put)
+                {
+                    return mirrorImpliedVol;
+                }
+                else if (Strike < UnderlyingPrice && Right == OptionRight.Call)
+                {
+                    return mirrorImpliedVol;
+                }
+                return impliedVol;
```

Comment:
I think we should add some comments what's/why this default implementation doing 🙏